runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -0,0 +1,438 @@
+      storeNode = TR::Node::create(TR::istore, 1, const1Node);
+      storeNode->setSymbolReference(anchoredNode->getSymbolReference());
+      }
+   else
+      TR_ASSERT_FATAL(false, "Anchord call has been turned into unexpected opcode %s\n", anchoredNode->getOpCode().getName());
+   prevBlock->append(TR::TreeTop::create(comp, storeNode));
+
+   // insert acmpeq for fastpath, taking care to set the proper register dependencies
+   // Any register dependencies added by splitPostGRA will now be on the BBExit for
+   // the call block.  As the ifacmpeq branching around the call block will reach the same
+   // target block, copy any GlRegDeps from the end of the call block to the ifacmpeq
+   auto* ifacmpeqNode = TR::Node::createif(TR::ifacmpeq, anchoredCallArg1TT->getNode()->getFirstChild(), anchoredCallArg2TT->getNode()->getFirstChild(), targetBlock->getEntry());
+   if (callBlock->getExit()->getNode()->getNumChildren() > 0)
+      {
+      TR::Node* glRegDeps = TR::Node::create(TR::GlRegDeps);
+      TR::Node* depNode = NULL;
+
+      if (anchoredNode->getOpCodeValue() == TR::iRegLoad)
+         {
+         depNode = TR::Node::create(TR::PassThrough, 1, storeNode->getChild(0));
+         depNode->setGlobalRegisterNumber(storeNode->getGlobalRegisterNumber());
+         glRegDeps->addChildren(&depNode, 1);
+         }
+
+      ifacmpeqNode->addChildren(&glRegDeps, 1);
+
+      TR::Node* expectedDeps = callBlock->getExit()->getNode()->getFirstChild();
+      for (int i = 0; i < expectedDeps->getNumChildren(); ++i)
+         {
+         TR::Node* temp = expectedDeps->getChild(i);
+         if (depNode && temp->getGlobalRegisterNumber() == depNode->getGlobalRegisterNumber())
+            continue;
+         else if (temp->getOpCodeValue() == TR::PassThrough)
+            {
+            // PassThrough nodes cannot be commoned because doing so does not
+            // actually anchor the child, causing it's lifetime to not be extended
+            TR::Node* original = temp;
+            temp = TR::Node::create(original, TR::PassThrough, 1, original->getFirstChild());
+            temp->setLowGlobalRegisterNumber(original->getLowGlobalRegisterNumber());
+            temp->setHighGlobalRegisterNumber(original->getHighGlobalRegisterNumber());
+            }
+         glRegDeps->addChildren(&temp, 1);
+         }
+      }
+
+   prevBlock->append(TR::TreeTop::create(comp, ifacmpeqNode));
+   }
+
+/*
+ * If value types are enabled, and the value that is being assigned to the array
+ * element might be a null reference, lower the ArrayStoreCHK by splitting the
+ * block before the ArrayStoreCHK, and inserting a NULLCHK guarded by a check
+ * of whether the array's component type is a value type.
+ */
+void
+TR::TreeLowering::lowerArrayStoreCHK(TR::Node *node, TR::TreeTop *tt)
+   {
+   traceMsg(comp(), "INVOKED!!!\n");
```

Comment:
Should be guarded with a `if (trace())` and given a more meaningful message.

Reply:
Fixed.